django/views/generic/detail.py

```diff
@@ -66,16 +66,26 @@ def get_queryset(self):
         may not be called if get_object() is overridden.
         """
         if self.queryset is None:
-            if self.model:
-                return self.model._default_manager.all()
+            if (model := self.get_model()) is not None:
+                return model._default_manager.all()
             else:
                 raise ImproperlyConfigured(
                     "%(cls)s is missing a QuerySet. Define "
                     "%(cls)s.model, %(cls)s.queryset, or override "
-                    "%(cls)s.get_queryset()." % {"cls": self.__class__.__name__}
+                    "%(cls)s.get_queryset() or %(cls)s.get_model()."
+                    % {"cls": self.__class__.__name__}
                 )
         return self.queryset.all()
 
+    def get_model(self):
```

Comment:
```suggestion
    def _get_model(self):
```
Do you think it's worth adding it to the public API? What do you think people will use it for? Shall we not just have this as an internal method?

Reply:
Thanks for the review. I will work on it this week.

Reply:
I don't imagine that overriding the method will be useful for people, and I'm all for making it private. However, given its role I think it may have to appear in the GCBVs method flowcharts. For example, the method flowchart for `UpdateView` wouldn't be accurate without it.

Let me know what you think.

(Sidenote 1: there are two open tickets that are about improving the method flowcharts: tickets [#24162](https://code.djangoproject.com/ticket/24162) and [#19227](https://code.djangoproject.com/ticket/19227))

(Sidenote 2: build failure now is due to a hanging ref to the _get_model() method which I removed from ref docs since it is now private. I'll fix it in the next update after a review.)